internal/testutils/endpoint.go

```diff
@@ -17,14 +17,39 @@ limitations under the License.
 package testutils
 
 import "github.com/kubernetes-incubator/external-dns/endpoint"
+import "sort"
 
 /** test utility functions for endpoints verifications */
 
+type byAllFields []*endpoint.Endpoint
+
+func (b byAllFields) Len() int      { return len(b) }
+func (b byAllFields) Swap(i, j int) { b[i], b[j] = b[j], b[i] }
+func (b byAllFields) Less(i, j int) bool {
+	if b[i].DNSName < b[j].DNSName {
+		return true
+	}
+	if b[i].DNSName == b[j].DNSName {
+		if b[i].Target < b[j].Target {
+			return true
+		}
+		if b[i].Target == b[j].Target {
+			if b[i].RecordType <= b[j].RecordType {
+				return true
+			}
+			return false
+		}
+		return false
+	}
+	return false
+}
+
 // SameEndpoint returns true if two endpoint are same
 // considers example.org. and example.org DNSName/Target as different endpoints
 // TODO:might need reconsideration regarding trailing dot
 func SameEndpoint(a, b *endpoint.Endpoint) bool {
```

Comment:
I think this should be an Equal method on top of endpoint.Endpoint. If you also do the Owner() method it is also simpler to read

```
func (ep Endpoint) Equal(other *endpoint.Endpoint) bool {
    return ep.DNSName == other.DNSName && ep.Target == other.Target && ep.RecordType == other.RecordType && ep.Owner() == other.Owner()
}
```